Add a pause button to the header when auto-refresh is enabled

rashidkpc opened the pull request:
This could in theory go into 4.1.0, but shouldn't block release, so I've added 4.2.0 to it

This adds a refresh button to the timepicker toggle of the navbar whenever auto-refresh is enabled. It:
- Should **not** fetch when clicking pause
- Should fetch when clicking play
- Should stop auto refresh while paused 
- Show the play button while paused
- Should restart auto refresh with the same interval when played
- Show the pause button while playing

![screen shot 2015-05-08 at 3 58 58 pm](https://cloud.githubusercontent.com/assets/1250387/7547079/2d6fec22-f59b-11e4-8c3d-7916673d9dc0.png)

Closes #3498

rashidkpc commented:
Closes #3498

spalger commented:
Hmm... now I see what you mean about clicking the refresh interval button.

Code and functionality looks great!

spalger commented:
Oh, and in case you didn't notice I took a stab at the tab switching and submit a PR against your fork.

w33ble commented:
I know this has been discussed at length, but I'm not sure where the discussion ended. Just wanted to chime in - this is how I use the timepicker, and now it doesn't work like I expect anymore (it doesn't close).

![2015-05-11 16_00_28](https://cloud.githubusercontent.com/assets/404731/7576809/b3ebe23a-f7f6-11e4-852d-b4b19a7e1581.gif)

w33ble commented:
I noticed that when you remove the refresh and add it back, it is added in "paused" mode. Is this intentional? 

![2015-05-11 16_04_06](https://cloud.githubusercontent.com/assets/404731/7576822/d2f78dfa-f7f6-11e4-95c4-80c0b2b747c6.gif)

rashidkpc commented:
Ok, fixed the pausing when disable/re-enable thing. I also tweaked the way tabs work:

Header in the new scenario:
- timepicker closed, refresh enabled: Show refresh interval, clicking it will open timepicker on refresh tab
- timepicker open, refresh enabled: Show refresh interval, clicking it will switch to refresh tab
- timepicker closed, refresh disabled: No autorefresh display, clicking time will open time picker
- timepicker open, refresh disabled: "Auto refresh" button shown, clicking it will switch to auto refresh display

I also got rid of the tabs in the dialog and added an arrow to indicate which tab was selected

![screen shot 2015-05-14 at 12 43 15 pm](https://cloud.githubusercontent.com/assets/1250387/7640251/f8506846-fa36-11e4-8b8c-ad8e43e5830f.png)

BigFunger commented:
There is an issue with visual artifacts when clicking on the legacy minimize button at the bottom of the roll out section instead of clicking again on the button at the top. 
![collapse artifact](https://cloud.githubusercontent.com/assets/12193008/7653699/634b5a88-fadd-11e4-9c1e-bc5c45ac4400.gif)

Also, if you modify the data range by clicking and dragging in the visualization area while the refresh is paused, the system does not update to the newly selected range.
